### PR TITLE
test(evmstaking/keeper): add test cases for withdraw

### PR DESCRIPTION
**Pull request by zsystm**
**coverage:**
- Increased test coverage from 0% to 86.8%
- **What's Covered**
  - Most application logics written on the Story side, excluding SDK.
- **What’s Not Covered (Edge Cases Related to the SDK)**
  - Failed to get validators (storage might broken)
  - Failed to get value from sdk store (e.g. `NextValidatorSweepIndex`)
  - etc...

**changes:**
- fixed error message
- added `accountKeeper`, `distributionKeeper` as test suite fields for writing new test codes using those keepers.

issue: #64

### CODE REVIEW

**Comment by limengformal** [comment]
@zsystm Can you fix the linter error?
https://github.com/piplabs/story/actions/runs/10579057061/job/29452157353?pr=65

File: client/x/evmstaking/keeper/withdraw_test.go [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+	for _, tc := range tcs {
+		s.Run(tc.name, func() {
+			if tc.settingMock != nil {
+				tc.settingMock()
+			}
+			err := keeper.ProcessWithdraw(ctx, tc.withdraw)
+			if tc.expectedErr != "" {
+				require.ErrorContains(err, tc.expectedErr)
+			} else {
+				require.NoError(err)
```
Comment (limengformal) [inline client/x/evmstaking/keeper/withdraw_test.go:433]: Can we check that [undelegate message](https://github.com/piplabs/story/blob/main/client/x/evmstaking/keeper/withdraw.go#L280) is sent?

File: client/x/evmstaking/keeper/withdraw_test.go [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+	for _, tc := range tcs {
+		s.Run(tc.name, func() {
+			if tc.settingMock != nil {
+				tc.settingMock()
+			}
+			err := keeper.EnqueueEligiblePartialWithdrawal(ctx, tc.input())
+			if tc.expectedError != "" {
+				require.ErrorContains(err, tc.expectedError)
+			} else {
+				require.NoError(err)
```
[the remark this answers is not in the document]
Comment (zsystm) [reply client/x/evmstaking/keeper/withdraw_test.go:317]: @limengformal 
applied at https://github.com/piplabs/story/pull/65/commits/bb0562f860275f14f19d01cadabc593869683d8f
- (added) check withdrawal is added to the queue for pass cases
- (updated) now, mock expectation reflects real values.
  - we can check token is burnt from delegator by mock expectation with delegator address

File: client/x/evmstaking/keeper/withdraw_test.go (hunk 1 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/withdraw_test.go:436]: @limengformal 
That's a good point 🙂 
I added check whether unbonding delegation is successfully created or not. (link. https://github.com/piplabs/story/pull/65/commits/d511d7e5200b41412e9bf6297e44296f2f07a42b)
If it is created, it means undelegate meesage sent and well processed.

**Review by limengformal — APPROVED** [review]
